tests/operator_test.go

```diff
@@ -2480,26 +2482,46 @@ spec:
 			// This label shouldn't exist, but this isn't harmful
 			// existing/running deployments will not be torn down until
 			// new ones are stood up (and the new ones will get stuck in scheduling)
+			var pods *k8sv1.PodList
 			labelKey := "kubevirt-test"
 			labelValue := "test-label"
 			infra := v1.ComponentConfig{
 				NodePlacement: &v1.NodePlacement{
 					NodeSelector: map[string]string{labelKey: labelValue},
 				},
 			}
+			By("Adding fake label to Virt components")
 			patchKvInfra(&infra, false, "")
-
-			Eventually(func() bool {
-				for _, name := range []string{"virt-controller", "virt-api"} {
+			for _, name := range []string{"virt-controller", "virt-api"} {
+				Eventually(func() error {
 					deployment, err := virtClient.AppsV1().Deployments(flags.KubeVirtInstallNamespace).Get(context.Background(), name, metav1.GetOptions{})
 					Expect(err).ToNot(HaveOccurred())
 					if deployment.Spec.Template.Spec.NodeSelector == nil || deployment.Spec.Template.Spec.NodeSelector[labelKey] != labelValue {
-						return false
+						return fmt.Errorf("NodeSelector should be propegated to the deployment eventually")
 					}
-				}
-				return true
-			}, 60*time.Second, 1*time.Second).Should(BeTrue())
+					return nil
+				}, 60*time.Second, 1*time.Second).ShouldNot(HaveOccurred())
 
+				Eventually(func() error {
+					pods, err = virtClient.CoreV1().Pods(flags.KubeVirtInstallNamespace).List(context.Background(),
+						metav1.ListOptions{
+							LabelSelector: fmt.Sprintf("kubevirt.io=%s", name),
+							FieldSelector: fields.ParseSelectorOrDie("status.phase=Pending").String(),
+						})
+					if err != nil {
+						return err
+					}
+					//The reason we check this is that sometime it takes a while until the pod is created and
+					//if the pod is created after the call to allPodsAreReady in the AfterEach scope
+					//than we will run the next test with side effect of pending pods of virt-api and virt-controller
+					//and increase flakiness
+					if len(pods.Items) == 0 {
+						errMsg := "the deployment should try to rollup the pods with the new selector and fail to schedule pods because the nodes don't have the fake label"
+						return fmt.Errorf(errMsg)
+					}
+					return nil
+				}, 60*time.Second, 1*time.Second).ShouldNot(HaveOccurred())
+			}
```

Comment:
Nice
Can we check instead that the deployment is ready ?
It will be nicer

something like
`./cluster/kubectl.sh wait deployment -n secondary secondary-dns --for=condition=Available --timeout=1m`
but with go code

Reply:
Please note that i don't wait for the deployment to be ready.
I'm waiting for a Pending pod to be created.

We need to make sure that the pod does exist otherwise if it will be created **AFTER** :
https://github.com/kubevirt/kubevirt/blob/b5c6c804eb60366e859147f253a964b207cd645a/tests/operator_test.go#L1047
we will run with side effects in the next test

Reply:
the pods won't be running because in this test they have fake node selector therefore the deployment will never be ready.